address.go

```diff
@@ -375,3 +483,82 @@ func (a *AddressPubKey) AddressPubKeyHash() *AddressPubKeyHash {
 func (a *AddressPubKey) PubKey() *btcec.PublicKey {
 	return a.pubKey
 }
+
+// AddressSegwit is an Address for a pay-to-witness-pubkey-hash (P2WPKH) or
```

Comment:
These should be split up into `AddressWitnessPubKeyHash` and `AddressWitnessScriptHash` in order to properly mirror the aspects of the prior PR that this one is replacing. With these changes, it should be able to slip in without any modifications to my prior PR.